File: 517-super-washing-machines/super-washing-machines.py
class Solution(object):
    def findMinMoves(self, machines):
        """
        :type machines: List[int]
        :rtype: int
        """
        sumM = sum(machines)
        
        if sumM % len(machines) != 0:
            return -1
        
        res = 0
        avg = sumM // len(machines)
        running_machine = 0

        for n in machines:
            num_mov = n - avg
            running_machine += num_mov
            res = max(res, abs(running_machine), num_mov)
        return res
        

File: 517-super-washing-machines/super-washing-machines.java
class Solution {
    public int findMinMoves(int[] machines) {
        int sumM = 0;
        for (int m : machines) {
            sumM += m;
        }
        
        if (sumM % machines.length != 0) {
            return -1;
        }
        
        int res = 0;
        int avg = sumM / machines.length;
        int runningMachine = 0;
        
        for (int n : machines) {
            int numMov = n - avg;
            runningMachine += numMov;
            res = Math.max(res, Math.max(Math.abs(runningMachine), numMov));
        }
        return res;
    }
}

